skiko/src/jvmMain/kotlin/org/jetbrains/skiko/FrameLimiter.kt

```diff
@@ -1,28 +1,28 @@
 package org.jetbrains.skiko
 
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.delay
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.*
+import kotlin.time.Duration.Companion.milliseconds
 import kotlin.time.ExperimentalTime
-
-private const val NanosecondsPerMillisecond = 1_000_000L
+import kotlin.time.TimeSource
 
 /**
  * Limit the duration of the frames (to avoid high CPU usage) to [frameMillis].
  * The actual delay depends on the precision of the system timer
  * (Windows has ~15ms precision by default, Linux/macOs ~2ms).
  * FrameLimiter will try to delay frames as close as possible to [frameMillis], but not greater
  */
+@OptIn(ExperimentalTime::class, ExperimentalCoroutinesApi::class)
```

Comment:
The same, we shouldn't use TimeSource here because it is experimental. At least it is in 1.8 (it is stabilized in 1.9). Not sure, if it is the same in 1.9, we need to compare, or just use the old `nanoTime`.

We can't upgrade Skiko to Kotlin 1.8, until we switch to it in Compose (this year approximately).